src/Workspaces/Core/Portable/Workspace/Host/DocumentService/DocumentPropertiesService.cs

```diff
@@ -16,5 +18,11 @@ internal class DocumentPropertiesService : IDocumentService
         /// but is not passed to the compiler when the containing project is built.
         /// </summary>
         public virtual bool DesignTimeOnly => false;
+
+        /// <summary>
+        /// The LSP client name that should get the diagnostics produced by this document; any other source
+        /// will not show these diagnostics. If null, the diagnostics do not have this special handling.
```

Comment:
can you expand on this (a lot).  what do you mean by "any other source will not show these diagnostics"?  who will show the diagnostics then?

Reply:
The error list would be the thing that this is primarly being excluded from. The idea is the Razor folks can query this and they're responsible for final display.

@dibarbet Do you want me to reword this in your PR or are you good?

Reply:
I should be able to